servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/NettyHttpServer.java

```diff
@@ -273,41 +271,39 @@ public String toString() {
                     connection.executionContext().ioExecutor(), connection.executionContext().executor(),
                     HttpExecutionStrategies.offloadNever());
             this.service = service;
-            // H2 uses child channels, doesn't support pipelining, and doesn't repeat the write operation on the same
-            // channel. We therefore don't need the splitting flush in this case.
-            if (protocol().major() <= 1) {
-                this.splittingFlushStrategy = new SplittingFlushStrategy(connection.defaultFlushStrategy(),
-                        new FlushBoundaryProvider() {
-                            private long contentLength;
-                            @Override
-                            public FlushBoundary detectBoundary(@Nullable final Object itemWritten) {
-                                if (itemWritten instanceof HttpResponseMetaData) {
-                                    final HttpResponseMetaData metadata = (HttpResponseMetaData) itemWritten;
-                                    contentLength = getContentLength(metadata);
-                                    // The content length maybe unknown at this point (e.g. 204 response) but then later
-                                    // determined to be 0. In that case we should conservatively use End and rely upon
-                                    // adjustForMissingBoundaries to accommodate if more data comes. Otherwise the
-                                    // FlushStrategy may not trigger a flush (e.g. flushOnEnd) and if so the response
-                                    // won't actually be written.
-                                    return contentLength > 0 ||
-                                            (contentLength < 0 && isTransferEncodingChunked(metadata.headers())) ?
-                                            Start : End;
-                                }
-                                if (itemWritten instanceof Buffer) {
-                                    return contentLength > 0 &&
-                                            (contentLength -= ((Buffer) itemWritten).readableBytes()) <= 0 ?
-                                            End : InProgress;
-                                }
-                                if (itemWritten instanceof HttpHeaders) {
-                                    return End;
-                                }
-                                return InProgress;
+            flushStrategy = new SplittingFlushStrategy(connection.defaultFlushStrategy(),
```

Comment:
I implemented a simplified version of `SplittingFlushStrategy` which doesn't care as much about switching, but the core concepts of lazy-set and boundaries are the same. figured it was best to avoid more complexity around flush strategy as the goal is to replace/remove `SplittingFlushStrategy` anyways.